### ISSUE TITLE
Inconsistent behavior with `View` in Rmarkdown knitting

### ISSUE THREAD

**Issue by 18kimn** [rstudio/rstudio#12243]
### System details

A Windows system and a Linux/MacOS system is necessary to fully reproduce this behavior. 

    RStudio Edition : Desktop
    RStudio Version : "Prairier Trillium" Release (1db809b8, 2022-05-20) for both systems
    OS Version: Microsoft Windows 10 Enterprise 10.0.19044; on Linux, NixOS (using nix 2.11.0, kernel version 5.15.74)
    R Version: 4.2.1 for both systems

### Steps to reproduce the problem

Create an Rmarkdown document with the following content. (Note: I'm using single quotation marks instead of backticks for the code chunks only because I can't figure out how to render verbatim Rmarkdown properly on GFM, sorry about that)
```
---
title: "test"
---

'''{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
'''

'''{r}
View(mtcars)
'''
```

And attempt to knit it by clicking the blue "Knit" button.

### Describe the problem in detail

`View` in a knitted Rmarkdown document appears to succeed in Windows but fail on Unix-based systems.  On my Linux system the render process throws an error which says "Error in .External2(C_dataviewer, x, title) : unable to start data viewer. Calls: <Anonymous> ... withVisible -> eval_with_user_handlers -> eval -> eval -> View. Execution halted`". On Windows no such error is thrown and the render succeeds, albeit without any output in the actual document

### Describe the behavior you expected

`View` should fail or succeed consistently on all types of systems; I'd prefer fail, as `View` has no meaning in a knitted document.

Apologies if this is the wrong repository to post this or if this does not qualify as a true bug !!

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
    - Sort of; I was able to do so on Linux, but I only have access to my school's Windows virtual desktop service which doesn't allow me to update system software
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by GegznaV** [comment]
As a workaround, for different viewing functions (e.g. `View()`, `help()`, `vignette()`, etc.), I recommend using chunk option `eval=FALSE`. Or excluding these functions from the script at all.

In case of failure, RStudio might recommend using this option (`eval=FALSE`) in an error message.

**Comment by 18kimn** [comment]
Yes, that is definitely the correct thing to do for the user, but it would be helpful to have it throw an error in all cases instead of only

**Comment by ronblum** [comment]
To add to that, on MacOS, XQuartz opens up and displays the object contents, though it's empty in the rendered object.

@jmcphers what is the expected behavior? Or, rather, what *should* it be?

**Comment by 18kimn** [comment]
please do not close this issue

**Comment by jgutman** [comment]
As a workaround, note that while using the Knit button with an Rmd file containing calls to `View()` has this annoying XQuartz behavior, if you Knit via the RStudio R Console by running `rmarkdown::render("my_file.Rmd", quiet=TRUE)`, your file renders appropriately, the View windows open in the RStudio data viewer, and XQuartz does not attempt to open. (Unfortunately this workaround does not appear to work for Quarto documents with the `quarto::quarto_render` command

We can consider getting this to work with the Knit push button as well

**Comment by 18kimn** [comment]
please do not close this issue

**Comment by GegznaV** [comment]
Not solved.

**Comment by GegznaV** [comment]
Is it solved?
